### PR TITLE
check50.internal.run_root_dir

### PR DESCRIPTION
**Pull request by Jelleas**
This PR introduces `check50.internal.run_root_dir` (effectively `check50.internal.run_dir.parent` while running a check) and `cd`s to that directory before importing the checks module.

This addresses two concerns:

1. Currently check50 does the first import of the checks module in the directory it was called from. That means that the following:

`__init__.py`
```Py
open("foo", "w")
```

Will create a file called `foo` in what is likely the directory the student is working from. Which is both unexpected and unwanted behavior.

2. Creates a persistent location for dynamically created files that need to exist across checks. For example in https://github.com/cs50/check50/issues/222#issuecomment-654267661 I used `check50.internal.run_root_dir` to store a list  of which checks need to exist, and each process then recreates the checks from that list.

### CODE REVIEW

**Comment by Jelleas** [comment]
`CheckRunner`  changed to a contextmanager that uses `lib50.working_area` and `lib50.cd` underneath the hood. This cleans up the code `__main__.py`  and makes  `CheckRunner.__init__` inexpensive to run.

**Comment by Jelleas** [comment]
Added `check50.internal.student_dir` that keeps track of the current working directory from where check50 was called. This makes it possible to access student files prior to running checks.

File: check50/internal.py [hunk 1]
```diff
@@ -13,12 +13,19 @@
 #: Directory containing the check and its associated files
 check_dir = None
 
-#: Temporary directory in which check is being run
+#: Temporary directory in which the current check is being run
 run_dir = None
 
+#: Temporary directory that is the root (parent) of all run_dir(s)
```
Comment (crossroads1112) [inline check50/internal.py:19]: Why is this necessary instead of just using run_dir.parent?

Comment (Jelleas) [reply check50/internal.py:19]: `run_dir` only gets set inside a check, that is after having imported the checks module. Whereas `run_root_dir` gets set before importing. This makes it possible to store things prior to running checks.
